org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/handler/ZigBeeCoordinatorHandler.java

```diff
@@ -230,40 +256,17 @@ public void initialize() {
             }
 
             try {
-                // Reset the initialization flag
+                // Reset the initialization flag and save the keys
                 Configuration configuration = editConfiguration();
+                configuration.put(CONFIGURATION_LINKKEY, linkKey.toString());
+                configuration.put(CONFIGURATION_NETWORKKEY, networkKey.toString());
                 configuration.put(CONFIGURATION_INITIALIZE, false);
                 updateConfiguration(configuration);
             } catch (IllegalStateException e) {
                 logger.error("Error updating configuration: Unable to reset initialize flag. ", e);
```

Comment:
Should this error log message be updated, because now the error could be caused by either resetting the initialize flag, or by updating the keys?

Suggestion: `logger.error("Error updating configuration with link key, network key, and initialize flag. ", e);`